### PR TITLE
Fixed issue where unsigned values were displayed as signed

### PR DESCRIPTION
**Pull request by Tupelov**
Closes #55

### CODE REVIEW

**Comment by Tupelov** [comment]
I may have messed up the decompilation output.
image

**Comment by Tupelov** [comment]
movie_draw makes perfect sense under this context and i've noticed that pointers now appear 0x10000 off inside the decompilation window when addiu is used to load them

**Comment by Tupelov** [comment]
Very weird though, because it looks like the instruction is loading the right place but the data is off by 0x10000, I've noticed this with function pointers as well. Maybe its me? I'm not sure

**Comment by Tupelov** [comment]
Might be a seperate issue as well

**Comment by Tupelov** [comment]
Doing this fixed it in all the places I've checked but it doesn't sit with me well.
image

File: data/languages/mips.sinc [hunk 1]
```diff
@@ -368,8 +368,8 @@ define pcodeop syscall;
     }
 
     # 0011 1100 000t tttt iiii iiii iiii iiii
-    :lui RT, immed                  		is prime=15 & rs=0 & RT & immed {
-        tmp:4 = immed << 16;
```
Comment (astrelsky) [inline data/languages/mips.sinc]: I think this one was correct. `lui` uses the unsigned immediate 16 bit value, shifts left 16 bits and sign extends it.

File: data/languages/mips.sinc [hunk 2]
```diff
@@ -550,8 +550,8 @@ define pcodeop syscall;
     }
 
     # 0010 11ss ssst tttt iiii iiii iiii iiii
-    :sltiu RT, RSsrc, simmed        		is prime=11 & RSsrc & RT & simmed {
-        RT = zext( RSsrc < simmed );
+    :sltiu RT, RSsrc, immed        		is prime=11 & RSsrc & RT & immed {
+        RT = zext( RSsrc < immed );
```
Comment (astrelsky) [inline data/languages/mips.sinc]: So according to the docs, the unsigned 16 `immed` value is sign extended and then the comparison is done as unsigned integers.

File: data/languages/mips.sinc [hunk 3]
```diff
@@ -666,8 +666,8 @@ define pcodeop syscall;
     }
 
     # 0000 01ss sss0 1001 iiii iiii iiii iiii
-    :tgeiu RSsrc, simmed            		is prime=1 & cond=9 & RSsrc & simmed {
-        if (RSsrc < simmed) goto <done>;
+    :tgeiu RSsrc, immed            		is prime=1 & cond=9 & RSsrc & immed {
+        if (RSsrc < immed) goto <done>;
```
Comment (astrelsky) [inline data/languages/mips.sinc]: Same as with `sltiu` above.

File: data/languages/mips.sinc [hunk 4]
```diff
@@ -696,8 +696,8 @@ define pcodeop syscall;
     }
 
     # 0000 01ss sss0 1011 iiii iiii iiii iiii
-    :tltiu RSsrc, simmed            		is prime=1 & cond=11 & RSsrc & simmed {
-        if (RSsrc >= simmed) goto <done>;
+    :tltiu RSsrc, immed            		is prime=1 & cond=11 & RSsrc & immed {
+        if (RSsrc >= immed) goto <done>;
```
Comment (astrelsky) [inline data/languages/mips.sinc]: Also same as above. Only adding the comment to be consistent.

**Review by astrelsky — CHANGES_REQUESTED** [review]

File: data/languages/mips.sinc (hunk 2 above)
Comment (Tupelov) [reply data/languages/mips.sinc]: Ahh so thats what it meant. Changed back

File: data/languages/mips.sinc (hunk 1 above)
Comment (Tupelov) [reply data/languages/mips.sinc]: Somehow it got broken, I have no idea how. Maybe it's my project?

**Comment by Tupelov** [comment]
I checked out lui in another project and it seemed fine so it's probably just me then. Still I think testing should be done just to be absolutely sure

**Comment by Tupelov** [comment]
Tested it out with a different version of the same game and it seem to be and SMT Nocturne specific issue somehow

**Comment by Tupelov** [comment]
But it doesn't affect the other game I tested Persona 3 FES

File: data/languages/mips.sinc (hunk 1 above)
Comment (Tupelov) [reply data/languages/mips.sinc]: yeah you were right

**Comment by Tupelov** [comment]
Ok so the weirdness was accidently making it unsigned when it was signed.

**Comment by astrelsky** [comment]
Don't worry about the failed checks. This was a mistake on my end and they only failed because I canceled them. The checks are for the java code so it isn't applicable here.

~~May you please squash the commits?~~ nvm

**Review by astrelsky — APPROVED** [review]

**Comment by JayFoxRox** [comment]
This PR was broken (as already described in #53 and #63)

>  i've noticed that pointers now appear 0x10000 off inside the decompilation window when addiu is used to load them

Here's an example which shows how such a problem can occur.

Here is the corrected disassembly:
```
ff 01 04 3c  	   lui        a0,0x1ff
00 00 84 24  	   _addiu     a0,a0,0x0
65 00 05 24  	   li         a1,0x65
01 00 06 3c  	   lui        a2,0x1
00 00 c6 24  	   addiu      a2,a2,0x0
00 f0 c6 24  	   addiu      a2,a2,0xfffff000 // This PR turned this into 0xF000 (which is incorrect)
db 76 10 0c  	   jal        memset
00 00 00 00  	   nop
```

Should be:
```c
uint32_t a0 = 0x1FF0000 + 0; // = 0x1FF0000
uint32_t a1 = 'A';
uint32_t a2 = 0x10000 + 0x0 - 0x1000; // = 0xF000
memset(a0, a1, a2);
```

Which is `memset(&_stack, 'A', _stack_size);`


With the code from this PR it becomes this broken piece; `memset(&_stack, 'A', 0x1f000);` (off by 0x10000):

```c
uint32_t a0 = 0x1FF0000 + 0; // = 0x1FF0000
uint32_t a1 = 'A';
uint32_t a2 = 0x10000 + 0x0 + 0xF000; // = 0x1F000
memset(a0, a1, a2);
```

**Pull request merged**